frontend/app/src/types/defi/compound.ts

```diff
@@ -1,34 +1,41 @@
-import { Balance, HasBalance } from '@rotki/common';
+import { Balance } from '@rotki/common';
+import { z } from 'zod';
 import {
   DEFI_EVENT_BORROW,
   DEFI_EVENT_LIQUIDATION,
   DEFI_EVENT_REPAY
 } from '@/services/defi/events';
 import { Collateral, CollateralizedLoan } from '@/types/defi/index';
 
-interface CompoundReward extends HasBalance {}
+const CompoundReward = z.object({
+  balance: Balance
+});
 
-type CompoundRewards = Readonly<Record<string, CompoundReward>>;
+const CompoundRewards = z.record(CompoundReward);
 
-interface CompoundLending extends HasBalance {
-  readonly apy: string | null;
-}
+const CompoundLending = z.object({
+  balance: Balance,
+  apy: z.string().nullish()
```

Comment:
`nullish()` is `| null | undefined`. 
Are you sure?

Reply:
I can play around a bit to see what the response is. We also need to go through the backend responses and see where it can be nullable or optional at some point.

In general, the backend returns either one or the other in most cases.

Reply:
It is nullable.